src/main/java/com/appland/appmap/output/v1/Parameters.java

```diff
@@ -38,58 +40,87 @@ public Parameters() { }
         javassist.bytecode.LocalVariableAttribute.tag);
 
+    // We should be able to handle methods without debug
+    // information. However, as of 20200822, other errors come up if
+    // do hook them, so bail out here.
     if (locals == null) {
-      throw new NoSourceAvailableException();
+      throw new NoSourceAvailableException("No local variables for " + fqn);
     }
 
-    Integer numberLocals = locals.tableLength();
-    CtClass[] parameterTypes = new CtClass[]{};
-
+    CtClass[] paramTypes = null;
     try {
-      parameterTypes = behavior.getParameterTypes();
+      paramTypes = behavior.getParameterTypes();
     } catch (NotFoundException e) {
-      Logger.println(
-          String.format("failed to get parameter types for %s.%s: %s",
-              behavior.getDeclaringClass().getName(),
-              behavior.getName(),
-              e.getMessage()));
+      throw new NoSourceAvailableException(
+        String.format("Failed to get parameter types for %s: %s",
+                      fqn, e.getMessage()));
     }
 
-    Boolean isStatic = (behavior.getModifiers() & Modifier.STATIC) != 0;
-    Value[] paramValues = new Value[parameterTypes.length];
-    for (int i = 0; i < numberLocals; ++i) {
-      // parameters are not neccesarily the first local variables
-      Integer paramIndex = locals.index(i);
-
-      if (!isStatic) {
-        // index 0 is `this` for nonstatic methods
-        // we don't need it
-        if (paramIndex == 0) {
-          continue;
+    String[] paramNames = null;
+    int numParams = paramTypes.length;
+    if (numParams > 0) {
+      int numLocals = locals.tableLength();
+      
+      // This is handy when debugging this code, put produces too much noise for general use.
+      if (false) {
+        Logger.println("local variables for " + fqn);
+        for (int idx = 0; idx < numLocals; idx++) {
+          Logger.printf("  %d %s %d\n", idx, locals.variableName(idx), locals.index(idx));
         }
+      }
 
-        // similarly, `parameterTypes` does not contain `this`, so shift our index back by one
-        paramIndex -= 1;
+      paramNames = new String[numParams];
+      Boolean isStatic = (behavior.getModifiers() & Modifier.STATIC) != 0;
+      int firstParamIdx = isStatic ? 0 : 1; // ignore `this`
+      int localVarIdx = 0;
+      
+      // Scan the local variables until we find the one with an index
+      // that matches the first parameter index.
+      //
+      // In some cases, though, there aren't local variables for the
+      // parameters. For example, the class file for
+      // org.springframework.samples.petclinic.owner.PetTypeFormatter,
+      // has the method
+      // print(Ljava/lang/Object;Ljava/util/Locale;)Ljava/lang/String
+      // in it, which only has a local variable for `this`. This
+      // method isn't in the source file, so I'm not sure where it's
+      // coming from.
```

Comment:
Maybe it’s created dynamically by Spring or something.

Reply:
Yeah, probably. There doesn't seem to be any way to generate a class file that has this issue.